### ISSUE TITLE
[LOVE] support more than 26 disks with virt resource

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#205]
This should be a beginner level patch for someone who is either new-ish to golang, or new to mgmt. It is meant to encourage new programmers to get involved with mgmt. It has a medium-level of algorithmic difficulty.

We have a virt resource in mgmt, As you can see here: https://github.com/purpleidea/mgmt/blob/cf2325a2daa1ba6df1f4095ed450e733eb110bd7/resources/virt.go#L1000 if we were to hit more than 26 disks, we'd have a numbering issue. If the 26th disk is sdz, the 27th should be sdaa, and then sdab, and so on till sdaz, then sdba, sdbb, sdbc, and so on...

Write a function to express this properly!

This patch would add the necessary additions, and optionally a small test or two in virt_test.go

If this takes you more than a few hours after you've got the hello world example building, or if you have any questions, please ping us in #mgmtconfig on Freenode IRC.

Thanks and happy hacking!

**Comment by bmbouter** [comment]
I want to try to do this issue

**Comment by purpleidea** [comment]
@bmbouter Awesome! Please ping if you have any questions. I once struggled with the a..b..aa..ab algorithm, so feel free to sneak a peak here if it makes you angry and you want a hint: https://github.com/purpleidea/vagrant-libvirt/commit/be2042537e4f8f3e59a7880bf3e09358252be252#diff-b34d2a33c102d0c3e664ab590a2119e7R3 )

**Comment by jonathangold** [comment]
Took a stab at this issue. Since I'm new to go and mgmt I wanted to run it by you before submitting a pull request. Here's a link to a playground demonstrating the function. 
https://play.golang.org/p/hOu0UGS1T1
Thoughts?

**Comment by purpleidea** [comment]
@jonathangold Hey, thanks for the function... You're welcome to send it as a PR, since this will get your familiar with our CI system, but in any case, I stripped out the XML part of your code, and ran a bit of a test:

https://play.golang.org/p/36m2umlRiv

As you can see after "z" it does not return the correct value.
So I think you're on the right track, but there's a bug.

**Comment by bmbouter** [comment]
@jonathangold I didn't find the time to do this so I'm glad you're contributing it!

**Comment by jonathangold** [comment]
For sure. On line 18 NumToAlpha(div) should be NumToAlpha(div - 1). Thought I had changed it.

**Comment by purpleidea** [comment]
That LGTM :)

Want to submit this (with a few tests for that function) as a patch?

**Comment by jonathangold** [comment]
When I try to run the tests both with and without my new test I get the following errors:
[vagrant@localhost resources]$ go test -run ''        
```
# _/home/vagrant/go/src/github.com/purpleidea/mgmt/resources
./nspawn.go:178: conn.GetProperties undefined (type *machine1.Conn has no field or method GetProperties)
./nspawn.go:188: conn.GetImage undefined (type *machine1.Conn has no field or method GetImage)
./nspawn.go:227: conn.TerminateMachine undefined (type *machine1.Conn has no field or method TerminateMachine)
FAIL	_/home/vagrant/go/src/github.com/purpleidea/mgmt/resources [build failed]
[vagrant@localhost resources]$ 

Any idea what's going on?
```

**Comment by purpleidea** [comment]
@jonathangold yes, when you cloned the repo you forgot to add the `--recursive` flag to git.

You can fix with `git submodule init && git submodule update` I think.

**Comment by jonathangold** [comment]
Didn't work. I'm using the vagrant image, so I don't think that's the issue.

**Comment by purpleidea** [comment]
@jonathangold can you `tree $GOPATH/src/github.com/purpleidea/mgmt/vendor/` ?
and `tree $GOPATH/src/github.com/coreos/`
?

**Comment by purpleidea** [comment]
Feel free to join us in #mgmtconfig on Freenode IRC for more back and forth help if you'd prefer.

**Comment by jonathangold** [comment]
I think the issue is because I was making changes in ~/go/... instead of ~/gopath/...

**Comment by purpleidea** [comment]
Ah, glad to hear it sounds like it's fixed :)
I think in golang 1.8 or 1.9+ if $GOPATH is not set it defaults to `~/go/`. But golang is picky about working inside of it. :P

**Comment by jonathangold** [comment]
submitted #211

**Comment by purpleidea** [comment]
Now fixed in 69f479b67e88f0d51dd11640a9d9accdd1aee129
